kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -302,6 +302,36 @@ public T apply(T resource) {
     }
   }
 
+
+  @Override
+  public D createOrReplaceWithNew() throws KubernetesClientException {
+    final Function<T, T> visitor = new Function<T, T>() {
+      @Override
+      public T apply(T resource) {
+        try {
+          return createOrReplace(resource);
+        } catch (Exception e) {
+          throw KubernetesClientException.launderThrowable(e);
+        }
+      }
+    };
+
+    try {
+      return getDoneableType().getDeclaredConstructor(Function.class).newInstance(visitor);
+    } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException | InstantiationException e) {
+      throw KubernetesClientException.launderThrowable(e);
+    }
+  }
+
+  @Override
+  public T createOrReplace(T item) {
+    if (get() == null) {
```

Comment:
@sayedasadali: It was intended to support `client.ResourceQuotas().inNamespace(namespace).createOrReplace(resourceQuota)`, If not, then its probably a bug.